### PR TITLE
TELCODOCS-82: RAN policies

### PR DESCRIPTION
**Pull request by StephenJamesSmith**
For version 4.8 Webscale Dev Preview

https://issues.redhat.com/browse/TELCODOCS-82

Direct link to doc preview: https://deploy-preview-34801--osdocs.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp-deploying-disconnected

Signed-off-by: Stephen Smith stesmith@redhat.com

### CODE REVIEW

File: modules/ztp-applying-the-ran-policies-for-monitoring-cluster-activity.adoc [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/ Module included in the following assemblies:
```
Comment (skrthomas) [inline modules/ztp-applying-the-ran-policies-for-monitoring-cluster-activity.adoc:1]: / Module included in the following assemblies: is appearing in the preview, FYI.

File: modules/ztp-applying-source-custom-resource-policies.adoc [hunk 2]
```diff
@@ -0,0 +1,159 @@
+/ Module included in the following assemblies:
```
Comment (skrthomas) [inline modules/ztp-applying-source-custom-resource-policies.adoc:1]: / Module included in the following assemblies: is appearing in the preview, FYI.

File: modules/ztp-generating-ran-policies.adoc [hunk 3]
```diff
@@ -0,0 +1,108 @@
+Module included in the following assemblies:
```
Comment (skrthomas) [inline modules/ztp-generating-ran-policies.adoc:1]: / Module included in the following assemblies: is appearing in the preview, FYI.

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Module included in the following assemblies:
```
Comment (skrthomas) [inline modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:1]: Included in preview.

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+//
+// scalability_and_performance/ztp-deploying-disconnected.adoc
+
+[id="ztp-things-to-consider-when-creating-custom-resource-policies_{context}"]
+= Things to consider when creating custom resource policies
```
Comment (skrthomas) [inline modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:6]: s/ Things to consider/ Considerations when creating...

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+* The custom resources used to create the ACM policies should be defined with consideration of possible overlay to its metadata and spec/data. For example, if the custom resource `metadata.name` does not change between clusters then you should set the `metadata.name` value in the custom resource file. If the custom resource will have multiple instances in the same cluster, then the custom resource `metadata.name` must be defined in the policy template file.
+
+* In order to apply the node selector for a specific machine config pool, you have to set the node selector value as `$mcp` in order to let the policy generator overlay the `$mcp` value with the defined mcp in the policy template.
+
+* Subscription source files will not be changed.
```
Comment (skrthomas) [inline modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:12]: s/ will not be/ do(es) not -->avoid future tense

File: modules/ztp-applying-source-custom-resource-policies.adoc [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+//
+// scalability_and_performance/ztp-deploying-disconnected.adoc
+
+[id="ztp-applying-source-custom-resource-policies_{context}"]
+= Applying source custom resource policies
+
+Source custom resource policies include:
```
Comment (skrthomas) [inline modules/ztp-applying-source-custom-resource-policies.adoc:8]: include the following: --> I've always been told to introduce a list with a complete sentence.

**Review by skrthomas — COMMENTED** [review]
A few comments for you and some suggestions. Nice work!! 💯

File: modules/ztp-applying-the-ran-policies-for-monitoring-cluster-activity.adoc (hunk 1 above)
Comment (StephenJamesSmith) [reply modules/ztp-applying-the-ran-policies-for-monitoring-cluster-activity.adoc:1]: Changed.

File: modules/ztp-applying-source-custom-resource-policies.adoc (hunk 2 above)
Comment (StephenJamesSmith) [reply modules/ztp-applying-source-custom-resource-policies.adoc:1]: Changed.

File: modules/ztp-generating-ran-policies.adoc (hunk 3 above)
Comment (StephenJamesSmith) [reply modules/ztp-generating-ran-policies.adoc:1]: Sigh. changed.

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc (hunk 4 above)
Comment (StephenJamesSmith) [reply modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:1]: Changed.

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc (hunk 5 above)
Comment (StephenJamesSmith) [reply modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:6]: Changed.

File: modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc (hunk 6 above)
Comment (StephenJamesSmith) [reply modules/ztp-things-to-consider-when-creating-custom-resource-policies.adoc:12]: Changed: "Subscription source files do not change."

File: modules/ztp-applying-source-custom-resource-policies.adoc (hunk 7 above)
Comment (StephenJamesSmith) [reply modules/ztp-applying-source-custom-resource-policies.adoc:8]: Change made.

File: modules/ztp-applying-source-custom-resource-policies.adoc [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  nodeSelector:
+    node-role.kubernetes.io/worker: ""
+  numVfs: $numVfs
+  priority: $priority
+  resourceName: $resourceName
+----
+
+.Example output
+
+The `SriovNetworkNodePolicy` name and `namespace` are the same for all clusters, so both are defined in the source `S`riovNetworkNodePolicy`.
```
Comment (juphoff) [inline modules/ztp-applying-source-custom-resource-policies.adoc:104]: Typo--extra backtick in: `S`riovNetworkNodePolicy`

File: modules/ztp-the-policygentemplate.adoc [hunk 9]
```diff
@@ -0,0 +1,132 @@
+/ Module included in the following assemblies:
```
Comment (juphoff) [inline modules/ztp-the-policygentemplate.adoc:1]: As mentioned for other files, this line is showing up in the preview.

File: modules/ztp-the-policygentemplate.adoc [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    #    - node-role.kubernetes.io/worker-du
+    #
+    # If the data is not defined, the defined data in the sourcePolicies/{fileName} will be carried out to
+    # the generated custom resource.
+    # If Any of the data items defined in the source file as variable start with $, it will be deleted
+    # from the generated data items if it is not setted.
+    data: "N/A"
+----
+
+The `group-du-ranGen.yaml` file defines a group of policies under a group named `group-du`. It defines a MachineConfigPool worker-du that is used as the node selector for any other policy defined in `sourceFiles`. An ACM policy is generated for every source file that exists in `sourceFiles`. And, a single placement binding and placement rule is generated to apply the cluster selection rule for group-du policies.
```
Comment (juphoff) [inline modules/ztp-the-policygentemplate.adoc:69]: Perhaps clarify beginning of second sentence with structure paralleling the end of the first?

It defines a MachineConfigPool named 'worker-du' that is used...

**Review by juphoff — APPROVED** [review]
Left a small number of minor-correction comments.

/lgtm

**Comment by StephenJamesSmith** [comment]
@ijolliffe All changes made. Please /lgtm if you haven't yet. Thx!

**Comment by StephenJamesSmith** [comment]
@vikram-redhat Distros: openshift-webscale added back to topic map, Final assembly file included. Please merge.

**Review by vikram-redhat — APPROVED** [review]

**Pull request merged**